imap_data_access/io.py

```diff
@@ -159,6 +163,85 @@ def query(
             "At least one query parameter must be provided. "
             "Run 'query -h' for more information."
         )
+    # Check instrument name
+    if instrument is not None:
+        if instrument not in (
+            "codice",
+            "glows",
+            "hi",
+            "hit",
+            "idex",
+            "lo",
+            "mag",
+            "swapi",
+            "swe",
+            "ultra",
+        ):
+            raise ValueError(
+                "Not a valid instrument, please choose from "
+                "('codice', 'glows', 'hi', 'hit', 'idex', "
+                "'lo', 'mag', 'swapi', 'swe', 'ultra')"
+            )
+
+    # Check data-level
+    # do an if statement that checks that data_level was passed in,
+    # then check it against all options, l0, l1a, l1b, l2, l3 etc.
+    if data_level is not None:
+        if data_level not in (
+            "l0",
+            "l1",
+            "l1a",
+            "l1b",
+            "l1c",
+            "l1ca",
+            "l1cb",
+            "l1d",
+            "l2",
+            "l2pre",
+            "l3",
+            "l3a",
+            "l3b",
+            "l3c",
+            "l3d",
+        ):
+            raise ValueError(
+                "Not a valid data level, choose from "
+                "('l0','l1','l1a','l1b','l1c','l1ca','l1cb',"
+                "'l1d','l2','l2pre','l3','l3a','l3b','l3c','l3d')."
+            )
+
+    # Check start-date
+    if start_date is not None:
+        try:
+            datetime.strptime(start_date, "%Y%m%d")
+        except ValueError as err:
+            raise ValueError("Not a valid start date, use format 'YYYYMMDD'.") from err
+
+    # Check end-date
+    if end_date is not None:
+        try:
+            datetime.strptime(end_date, "%Y%m%d")
+        except ValueError as err:
+            raise ValueError("Not a valid end date, use format 'YYYYMMDD'.") from err
+
+    # Check version make sure to include 'latest'
+    if version is not None:
```

Comment:
Similarly, could you put this check in some kind of `is_valid_version` in ScienceFilePath so we can share it around?